static/js/directives/enter.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Directive to detect enter key
+ * */
+angular.module('inboxDirectives').directive('ngEnter', function($document) {
+  'use strict';
+  return {
+    restrict: 'A',
+    link: function(scope, elem, attrs) {
+      var keydownEvent = function(event) {
+        if (event.which === 13) {
+          scope.$eval(attrs.ngEnter);
+          event.preventDefault();
+          $document.unbind('keydown keypress', keydownEvent);
+        }
+      };
+      $document.bind('keydown keypress', keydownEvent);
```

Comment:
I think you also need to unbind if the modal is cancelled. This should work...

```js
scope.$on('$destroy', function() {
  $document.unbind('keydown keypress', keydownEvent);
});
```

This will need some testing so use console statements or breakpoints to ensure the listeners are being cleaned up correctly.